src/lib.rs

```diff
@@ -526,34 +534,33 @@ where
 }
 
 #[cfg(feature = "cpp_demangle")]
-fn demangle_cpp_symbol(mangled: &std::ffi::CStr) -> Cow<str> {
-    if let Ok(sym) = cpp_demangle::Symbol::new(mangled.to_bytes()) {
+fn demangle_cpp_symbol(mangled: &[u8]) -> Cow<str> {
+    if let Ok(sym) = cpp_demangle::Symbol::new(mangled) {
         Cow::from(format!("{}", sym))
     } else {
-        mangled.to_string_lossy()
+        String::from_utf8_lossy(mangled)
     }
 }
 
 #[cfg(not(feature = "cpp_demangle"))]
-fn demangle_cpp_symbol(mangled: &std::ffi::CStr) -> Cow<str> {
-    mangled.to_string_lossy()
+fn demangle_cpp_symbol(mangled: &[u8]) -> Cow<str> {
+    String::from_utf8_lossy(mangled)
 }
 
 #[cfg(feature = "rustc-demangle")]
-fn demangle_rust_symbol(mangled: &std::ffi::CStr) -> Cow<str> {
+fn demangle_rust_symbol(mangled: &[u8]) -> Cow<str> {
     Cow::from(format!(
         "{}",
-        rustc_demangle::demangle(mangled.to_string_lossy().as_ref())
+        rustc_demangle::demangle(String::from_utf8_lossy(mangled).as_ref())
     ))
 }
 
 #[cfg(not(feature = "rustc-demangle"))]
-fn demangle_rust_symbol(mangled: &std::ffi::CStr) -> Cow<str> {
-    mangled.to_string_lossy()
+fn demangle_rust_symbol(mangled: Cow<str>) -> Cow<str> {
+    String::from_utf8_lossy(mangled)
```

Comment:
Actually, the problem is I got the argument type wrong (and didn't build without rustc-demangle to test it).